twake/frontend/src/app/components/OnBoarding/AccountStatusComponent.tsx

```diff
@@ -6,27 +6,48 @@ import Languages from 'services/languages/languages.js';
 import ModalManager from 'app/components/Modal/ModalManager';
 import UnverifiedAccount from './popups/UnverifiedAccount';
 import BlockedAccount from './popups/BlockedAccount';
+import UserService from 'services/user/user.js';
 
 const AccountStatusComponent = (): JSX.Element => {
-  const [period, setPeriod] = useState<number>(8);
-  const [displayBanner, setDisplayBanner] = useState<boolean>(
-    period < 7 && period > 0 ? true : false,
-  );
+  const user = UserService.getCurrentUser();
+  const [period, setPeriod] = useState<number>(0);
+  const [displayBanner, setDisplayBanner] = useState<boolean>(false);
+  const periodLimit: number = 7;
 
   useEffect(() => {
+    calculateTrialPeriod();
     isBlockedAccount();
-  }, [period]);
+  }, [period, displayBanner]);
+
+  const calculateTrialPeriod = (): void => {
+    const oneDay: number = 1000 * 60 * 60 * 24;
+    // user created_at
+    const createdDay: number = user.created_at;
+    const currentDay: number = Date.now();
+    const shouldDisplayBanner: boolean = period > 0 && period < periodLimit ? true : false;
+    const currentPeriod: number = Math.round(Math.round(currentDay - createdDay) / oneDay);
+
+    if (!user.is_verified) {
+      setPeriod(currentPeriod);
+      return setDisplayBanner(shouldDisplayBanner);
+    }
+  };
+
+  const truncateDaysLeft = (period: number) => Math.trunc(period);
```

Comment:
You can simplify all this a lot and remove all useStates and useEffects:
```
const maxUnverifiedDays: number = 7;

const user = UserService.getCurrentUser();
const oneDay: number = 1000 * 60 * 60 * 24;
const periodLimit: number = (user.created_at || 0) + maxUnverifiedDays * oneDay;
const daysLeft: number = Math.ceil( ( ( periodLimit - Date.now() ) / oneDay );

const showBanner = !user.is_verified;
if(!user.is_verified && daysLeft <= 0){
   showBlockedModal();
}
```

Reply:
done